project-koku/koku
# [COST-2371] Azure Query for HCS

Fixes [COST-2371](https://issues.redhat.com/browse/COST-2371)

Changes proposed in this PR:
* Allow HCS processing to work for AWS or Azure and include date filtering for Azure SQL
* Adjust the trino SQL query process to allow for the description to be returned to build the csv header column list from the query

Testing Instructions:
1. Spin up koku locally with Trino, such as: `make docker-up-min-trino`
2. After everything spins up, ingest some AWS data:
      - `make create-test-customer`
      - `make test_source=aws load-test-customer-data`
3. Ingest some Azure data:
      - `make test_source=azure load-test-customer-data`
4. Check minio to see that you have no HCS data in the koku-bucket (there would be a folder called HCS next to the data one) http://localhost:9090/login
5.  Add your account to unleash under the `hcs-data-processor` http://localhost:4242/#/features/strategies/hcs-data-processor
6. Hit the masu endpoint for hcs and put in your AWS source info:  http://localhost:5042/api/cost-management/v1/hcs_report_data/?start_date=2022-04-01&schema=acct10001&provider_type=AWS&provider_uuid=02d80185-78ea-45c5-b75b-925786d26b7d

7. Check minio again and this time there is a path down HCS that leads to some daily AWS csvs based on the date you used int he masu endpoint
8. Repeat 6 and 7 but for the Azure V2 source


---
Additional Context:
[Smokes](https://ci.ext.devshift.net/job/project-koku-koku-pr-check/4760/), these are being investigated with our stage issues so not related to this PR: 
```
Test Result (2 failures / -1615)tests.rest_api.v1.test_source.test_api_ocp_on_aws_source_crudtests.rest_api.v1.test_source.test_api_ocp_on_azure_source_crud
--
[Test Result](https://ci.ext.devshift.net/job/project-koku-koku-pr-check/4760/testReport/) (2 failures / -1615)
[tests.rest_api.v1.test_source.test_api_ocp_on_aws_source_crud](https://ci.ext.devshift.net/job/project-koku-koku-pr-check/4760/testReport/junit/tests.rest_api.v1/test_source/test_api_ocp_on_aws_source_crud/)
[tests.rest_api.v1.test_source.test_api_ocp_on_azure_source_crud](https://ci.ext.devshift.net/job/project-koku-koku-pr-check/4760/testReport/junit/tests.rest_api.v1/test_source/test_api_ocp_on_azure_source_crud/)
```
